rcl/include/rcl/event.h

```diff
@@ -0,0 +1,165 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__EVENT_H_
+#define RCL__EVENT_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl/client.h"
+#include "rcl/macros.h"
+#include "rcl/publisher.h"
+#include "rcl/service.h"
+#include "rcl/subscription.h"
+#include "rcl/visibility_control.h"
+
+
+typedef enum rcl_publisher_event_type_t
+{
+  RCL_PUBLISHER_DEADLINE,
+  RCL_PUBLISHER_LIVELINESS
+} rcl_publisher_event_type_t;
+
+
+typedef enum rcl_subscription_event_type_t
+{
+  RCL_SUBSCRIPTION_DEADLINE,
+  RCL_SUBSCRIPTION_LIVELINESS
+} rcl_subscription_event_type_t;
+
+typedef enum rcl_client_event_type_t
+{
+  RCL_CLIENT_EVENT_UNIMPLEMENTED
+} rcl_client_event_type_t;
+
+typedef enum rcl_service_event_type_t
+{
+  RCL_SERVICE_EVENT_UNIMPLEMENTED
+} rcl_service_event_type_t;
+
+
+/// Internal rcl implementation struct.
+struct rcl_event_impl_t;
+
+/// Structure which encapsulates a ROS Subscription.
```

Comment:
Comment looks wrong for the type. After reviewing a lot of other code it seems like the name of this is misleading as it isn't actually an occurrence of an event but rather some sort of event handler. I think this needs a better name and documentation.

Reply:
I think `rcl_subscription_t` and `rcl_publisher_t` has the same issue.

Anyways, will fix this comment. Thanks for catching it.